src/Credentials/GCECredentials.php

```diff
@@ -235,7 +243,19 @@ public function fetchAuthToken(callable $httpHandler = null)
             return array();  // return an empty array with no access token
         }
 
-        $json = $this->getFromMetadata($httpHandler, self::getTokenUri());
+        $tokenUri = self::getTokenUri();
+        if ($this->scope) {
```

Comment:
WDYT about making sure we don't need to execute this computation every time `fetchAuthToken` is called?

Reply:
Sure, i could move it to a property created in the constructor.

Reply:
Sounds good to me

Reply:
Done, PTAL!